test/test.py

```diff
@@ -946,7 +971,7 @@ def testAggregations(self):
             self.assertEqual('RediSearch', res[23])
             self.assertEqual(2, len(res[25]))
 
-    def testIndexDefiniontion(self):
+    def testIndexDefinition(self):
```

Comment:
Might as well docstring while you're here. 

Also, I'd split tests into smaller samples - one for IndexType.JSON and one for IndexType.HASH.  It feels more pythonic to have (many) smaller tests. This last one is a style issue, not a requirement.